x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/saml/TransportSamlInvalidateSessionAction.java

```diff
@@ -102,44 +104,30 @@ private void findAndInvalidateTokens(SamlRealm realm, SamlLogoutRequestHandler.R
         final Map<String, Object> tokenMetadata = realm.createTokenMetadata(result.getNameId(), result.getSession());
         if (Strings.isNullOrEmpty((String) tokenMetadata.get(SamlRealm.TOKEN_METADATA_NAMEID_VALUE))) {
             // If we don't have a valid name-id to match against, don't do anything
-            logger.debug("Logout request [{}] has no NameID value, so cannot invalidate any sessions", result);
+            LOGGER.debug("Logout request [{}] has no NameID value, so cannot invalidate any sessions", result);
             listener.onResponse(0);
             return;
         }
 
         tokenService.findActiveTokensForRealm(realm.name(), containsMetadata(tokenMetadata), ActionListener.wrap(tokens -> {
-            logger.debug("Found [{}] token pairs to invalidate for SAML metadata [{}]", tokens.size(), tokenMetadata);
+            LOGGER.debug("Found [{}] token pairs to invalidate for SAML metadata [{}]", tokens.size(), tokenMetadata);
             if (tokens.isEmpty()) {
                 listener.onResponse(0);
             } else {
-                GroupedActionListener<TokensInvalidationResult> groupedListener = new GroupedActionListener<>(
-                    tokens.size(),
-                    ActionListener.wrap(collection -> listener.onResponse(collection.size()), listener::onFailure)
-                );
-                tokens.forEach(tuple -> invalidateTokenPair(tuple, groupedListener));
+                tokenService.invalidateAllTokens(tokens, ActionListener.wrap(tokensInvalidationResult -> {
+                    if (LOGGER.isInfoEnabled() && tokensInvalidationResult.getErrors().isEmpty() == false) {
+                        try (XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent())) {
+                            tokensInvalidationResult.toXContent(builder, ToXContent.EMPTY_PARAMS);
+                            LOGGER.info("Failed to invalidate some SAML access or refresh tokens {}", Strings.toString(builder));
```

Comment:
I opted to keep the same log level that's been used in `invalidateAccessToken`.
If this was a case of introducing new logging, I would've probably opted for WARN myself too, but it makes little difference since both WARN and INFO levels are enabled by default.